packages/core-web/src/index.js

```diff
@@ -42,34 +42,61 @@ function detectAndApplyFixedHeaderStyles() {
 
   const headerHeight = headerSelector.height();
   const bufferHeight = 1;
-  insertCss(`.fixed-header-padding { padding-top: ${headerHeight}px !important }`);
+  insertCss(`.fixed-header-padding { padding-top: ${headerHeight}px !important; }`);
   insertCss(
     `span.anchor {
     position: relative;
-    top: calc(-${headerHeight}px - ${bufferHeight}rem)
+    top: calc(-${headerHeight}px - ${bufferHeight}rem);
     }`,
   );
   insertCss(`.nav-menu-open { max-height: calc(100% - ${headerHeight}px); }`);
+  insertCss(`.DocSearch-Modal { margin-top: ${headerHeight}px; }`);
+  insertCss(
+    `@media (max-width: 750px) {
+      .DocSearch-Dropdown {
+        height: calc(100% - ${headerHeight}px - 66px);
+      }
+    }`,
+  );
```

Comment:
Just a quick look, would it be better to keep these strictly to the plugin itself? Since docsearch isn't really core functionality, but a plugin.

Some questions release wise:
- whether its possible to deliver a more minimal fix by adding the `apiKey` you mentioned onto `v3.1.1` or so (also so no immediate action needed on author end to migrate v2 -> v3)
- while we merge this bigger change into the main branch for `v4.0` (since v2 -> v3 docsearch is a breaking change)

Reply:
> Just a quick look, would it be better to keep these strictly to the plugin itself? Since docsearch isn't really core functionality, but a plugin.

I've shifted the plugin css code to within the plugin file. The rest of the plugin code related to `adjustHeaderClasses` is left there to minimise duplication as there are a couple of conditions that will need to adjust the plugin's css.

> * whether its possible to deliver a more minimal fix by adding the apiKey you mentioned onto v3.1.1 or so (also so no immediate action needed on author end to migrate v2 -> v3)
>
> * while we merge this bigger change into the main branch for v4.0 (since v2 -> v3 docsearch is a breaking change)

Yup agree with this. We should do the v2 -> v3 docsearch in our v4.0 and release the minimal fix in v3.1.1. I'll create a separate branch called `v3.1.x-hotfixes` (similar to the v2.7.x-hotfixes branch) and push the update there. Is this workflow fine?

Reply:
> Yup agree with this. We should do the v2 -> v3 docsearch in our v4.0 and release the minimal fix in v3.1.1. I'll create a separate branch called v3.1.x-hotfixes (similar to the v2.7.x-hotfixes branch) and push the update there. Is this workflow fine?

👍 

Alternatively there haven't been any breaking changes since the last release so far, so we could also merge the more minimal fix directly on the main branch. (I think either is fine)

Reply:
> I've shifted the plugin css code to within the plugin file. The rest of the plugin code related to adjustHeaderClasses is left there to minimise duplication as there are a couple of conditions that will need to adjust the plugin's css.


> V3 has changed its design drastically from V2. The search bar has a new design and clicking on the search bar will open a modal instead. The searches will be done on this modal and the results will be displayed on the modal instead of the dropdown that is used by Markbind's search. Since the new search bar and modal are auto generated by Algolia's package, it will be complicated to try to make the design similar to the current Markbind's search. Is it fine to stick to their native design?


Sorry, I completely missed the purpose of these changes earlier.

Are the dynamic styles added to tackle the above mentioned issue?

I'm favouring sticking with algolia's default design as much as possible if so, imo there isn't much benefit from pushing the dropdown / ui downward. Coupling our dynamic style detections over to an external element might break as well if algolia's styles / classes / dom structure changes.

If its an issue with correctness (e.g. the modal going out of the screen) I'll need a little more time to review this with a v3 algolia setup

Reply:
Mainly because the modal gets blocked by the navbar if the header is fixed.

![image](https://user-images.githubusercontent.com/47470981/150630481-1336d3d5-d233-44a8-8fd6-0c0e2cf30dd1.png)

I could try to increase the `z-index` of the modal to be higher. That may help to resolve this issue without pushing it down below the fixed header.

Reply:
Tested with z-index and it works as well. Will change to this implementation as it's much simpler.

![image](https://user-images.githubusercontent.com/47470981/150631043-153fb402-0c2f-45ff-9fc0-87d80874cb06.png)